modules/ROOT/pages/errors/gql-errors/22ND2.adoc

```diff
@@ -0,0 +1,59 @@
+= 22ND2
+
+== Status description
+error: data exception - operation not allowed for roles with DENY privileges. Invalid input: '{ <<query>> }' is not allowed for roles with DENY privileges.
+
+== Explanation
+If an auth rule fails to evaluate, for example because it depends on a claim that the users auth token does not have, Neo4j will default to not give the role to the user.
+
+When a role contains denied privileges, not giving it to the user leads to elevated privileges. In order to avoid this, roles are prohibited from having denied privileges and be granted to an auth rule at the same time.
```

Comment:
```suggestion
When a role contains denied privileges, not assigning the role actually removes the denies, which means the user ends up with more privileges than they should.
To avoid this, roles cannot have both `DENY` privileges and be granted to an auth rule.
```